runtime/shared_common/ScopeManagerImpl.cpp

```diff
@@ -398,24 +399,32 @@ SH_ScopeManagerImpl::validate(J9VMThread* currentThread, const J9UTF8* partition
 	cachedModContext = findScopeForUTF(currentThread, modContext);
 	scopedMatch = (ScopedROMClassWrapper*)ITEMDATA(match);
 	if (cachedPartition) {
-		if (!(scopedMatch->partitionOffset && (cachedPartition==(const J9UTF8*)RCWPARTITION(scopedMatch)))) {
+		const J9UTF8* partition = (const J9UTF8*)_cache->getAddressFromJ9ShrOffset(&(scopedMatch->partitionOffset));
+		if ((NULL == partition)
+			|| (cachedPartition != partition)
+		) {
 			Trc_SHR_SMI_validate_ExitFailed2(currentThread);
 			return 0;
 		}
-	} else 
-	if (scopedMatch->partitionOffset) {
-		Trc_SHR_SMI_validate_ExitFailed3(currentThread);
-		return 0;
+	} else {
+		if (scopedMatch->partitionOffset.offset) {
```

Comment:
`offset` is not a boolean; this should be `if (0 != scopedMatch->partitionOffset.offset)`.

Reply:
Fixed